client/containers/Pub/PubDocument/PubWordCountButton.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useMemo, useState } from 'react';
+import { Popover, Button, Icon } from '@blueprintjs/core';
+
+import { Node } from 'prosemirror-model';
+
+require('./pubWordCountButton.scss');
+
+type Props = {
+	doc: Node;
+};
+
+const getWordAndCharacterCountsFromDoc = (node: Node) => {
+	const text = node.textBetween(0, node.content.size, ' ', ' ');
+	const words = text.split(' ').filter((word) => word !== '');
+	const wordCount = words.length;
+	const characterCount = words.reduce((a, x) => a + x.length, 0);
+	return [wordCount, characterCount];
+};
+
+const PubHeaderFormattingWordCountButton = (props: Props) => {
+	const { doc } = props;
+	const [wordCount, characterCount] = useMemo(() => getWordAndCharacterCountsFromDoc(doc), []);
```

Comment:
This should have `[doc]` as its deps array, right? For performances' sake, I recommend doing something like this:

```ts
const counts = useMemo(() => {
    if (open) {
        return getWordAndCharacterCountsFromDoc(doc);
    }
    return null;
}, [open, doc]);
```

that way we're really only computing this when the popover is open, which seems most ideal.

Reply:
Yep, great suggestion. I'll get this in now.